src/Network/Ethereum/Core/Signatures.purs

```diff
@@ -98,9 +101,25 @@ newtype Address = Address HexString
 derive newtype instance addressShow :: Show Address
 derive newtype instance addressEq :: Eq Address
 derive newtype instance addressOrd :: Ord Address
-derive newtype instance decodeAddress :: Decode Address
 derive newtype instance encodeAddress :: Encode Address
 
+instance decodeAddress :: Decode Address where
+  decode a = do
+    hxString <- decode a
+    case mkAddress hxString of
+      Nothing -> fail <<< ForeignError $ "Address must be 20 bytes long: " <> show hxString
+      Just res -> pure res
+
+instance decodeJsonAddress :: A.DecodeJson Address where
+  decodeJson json = do
+    hxString <- A.decodeJson json
+    case mkAddress hxString of
+      Just res -> pure res
+      Nothing -> Left $ "Address must be 20 bytes long: " <> show hxString
+
+instance encodeJsonAddress :: A.EncodeJson Address where
```

Comment:
There is no Encode instance, not sure if it's relevant here.

encode_ decode_?